### DISCUSSION TITLE
Best way to test queue durability?

### DISCUSSION THREAD

**Discussion by johnnyleejy** [taskforcesh/bullmq#1460] [Q&A] [answered]
I'm currently trying to test a queue's durability and my test case involves restarting the queue to ensure messages are not lost.

What's the best way to test bullmq's durability for this test case.

I tried to disconnect a queue but there doesn't seem to be any method allowing me to restart me queue or worker again.

Thank you :)

**Comment by johnnyleejy** [comment, marked as answer]
Here's what I did.

I created listeners to listen if the redisclient did disconnect and reconnect.

Then I returned promises and expect a true value in my tests.
   
    const queueClient = await queue.client;
    const didQueueDisconnect = new Promise<boolean>((resolve) => {
      setTimeout(resolve, 10000, false); // resolves to false if client is not closed after 10 seconds
      queueClient.on('close', () => {
        resolve(true); // resolves to true when queue connection is closed
      });
    });
    const didQueueReconnect = new Promise<boolean>((resolve) => {
      setTimeout(resolve, 10000, false); // resolves to false if client is not reconnected after 10 seconds
      queueClient.on('reconnecting', () => {
        resolve(true); // resolves to true when queue is reconnecting
      });
    });
    const isQueueReady = new Promise<boolean>((resolve) => {
      setTimeout(resolve, 10000, false); // resolves to false if client is not ready after 10 seconds
      queueClient.on('ready', () => {
        resolve(true); // resolves to true when queue is ready for commands
      });
    });

    queueClient.disconnect(true);
    expect(await didQueueDisconnect).toBeTrue(); // Queue was disconnected
    expect(await didQueueReconnect).toBeTrue(); // Queue attempted to reconnect
    expect(await isQueueReady).toBeTrue(); // Queue is ready to receive commands
    expect(queueClient.status).toStrictEqual('ready');

**Reply by manast** [reply]
I am not sure what you mean with durability, but the connection to redis is handled by ioredis and by default it will retry on reconnections, you do not need any other logic than the correct options for ioredis that do what you want: https://github.com/luin/ioredis#auto-reconnect
